YAML/YAML.sublime-syntax

```diff
@@ -620,47 +704,36 @@ contexts:
     - include: block-pair
 
   block-pair:
-    - match: \?
-      scope: punctuation.definition.key-value.begin.yaml
+    - match: \?(?=\s|$)
+      scope: meta.mapping.yaml punctuation.definition.key-value.begin.yaml
       push:
-        - meta_scope: meta.block-mapping.yaml
-        - match: (?=\?) # Empty mapping keys & values are allowed
-          pop: true
-        - match: ^ *(:)
-          captures:
-            1: punctuation.separator.key-value.mapping.yaml
-          pop: true
-        - match: ':'
-          scope: invalid.illegal.expected-newline.yaml
-          pop: true
-        - include: block-node
-    # Attempt to match plain-out scalars and highlight as "entity.name.tag",
+        - block-key-explicit
```

Comment:
Maybe use a direct `push: block-key-explicit` as a single context is pushed only?